File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/controller/ProductController.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.controller;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ProductDetailResponse;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ProductResponse;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.service.ProductService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api")
public class ProductController {

    private final ProductService productService;

    @GetMapping("/v1/products")
    public ResponseEntity<List<ProductResponse>> getProducts(
            @RequestParam(required = false) String search,
            @RequestParam(required = false) Boolean certified,
            @RequestParam(required = false) String grade,
            @RequestParam(required = false) List<String> categoryId) {
        return ResponseEntity.ok(
                productService.getProducts(search, certified, grade, categoryId));
    }

    @GetMapping("/v1/products/{productId}")
    public ResponseEntity<ProductDetailResponse> getProductDetail(
            @PathVariable String productId) {
        return ResponseEntity.ok(
                productService.getProductDetail(UUID.fromString(productId)));
    }
}

File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/controller/ProductController.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.controller

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ProductDetailResponse
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ProductResponse
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.service.ProductService
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/api/v1/products")
class ProductController(
    private val productService: ProductService
) {

    @GetMapping("/")
    fun getProducts(
        @RequestParam(required = false) search: String?,
        @RequestParam(required = false) certified: Boolean?,
        @RequestParam(required = false) grade: String?,
        @RequestParam(required = false) categoryId: List<String>?
    ): List<ProductResponse> {
        return productService.getProducts(search, certified, grade, categoryId)
    }

    @GetMapping("/{productId}")
    fun getProductDetail(
        @PathVariable productId: String
    ): ProductDetailResponse {
        return productService.getProductDetail(UUID.fromString(productId))

    }
}

